osu.Game/Overlays/Settings/Sections/SkinSection.cs

```diff
@@ -59,15 +65,32 @@ private void load(OsuConfigManager config, SkinManager skins)
 
             config.BindWith(OsuSetting.Skin, configBindable);
 
+            usableSkins = skins.GetAllUsableSkins().ToArray();
+
             skinDropdown.Bindable = dropdownBindable;
-            skinDropdown.Items = skins.GetAllUsableSkins().ToArray();
+            skinDropdown.Items = usableSkins.Length > 1 ? usableSkins.Concat(new[] { random_skin_info }) : usableSkins;
 
             // Todo: This should not be necessary when OsuConfigManager is databased
             if (skinDropdown.Items.All(s => s.ID != configBindable.Value))
                 configBindable.Value = 0;
 
             configBindable.BindValueChanged(v => dropdownBindable.Value = skinDropdown.Items.Single(s => s.ID == v), true);
-            dropdownBindable.BindValueChanged(v => configBindable.Value = v.ID);
+            dropdownBindable.BindValueChanged(v =>
+            {
+                if (v == random_skin_info)
+                    randomizeSkin();
+                else
+                    configBindable.Value = v?.ID ?? 0;
```

Comment:
it there a reason for this null check? it wasn't there previously, so curious.

Reply:
without the null check it crashed when the dropdown menu was open while clicking on the yellow reset button to the left of it.